homeassistant/components/hive/config_flow.py

```diff
@@ -0,0 +1,184 @@
+"""Config Flow for Hive."""
+
+from apyhiveapi import Auth
+from apyhiveapi.helper.hive_exceptions import (
+    HiveApiError,
+    HiveInvalid2FACode,
+    HiveInvalidPassword,
+    HiveInvalidUsername,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_SCAN_INTERVAL, CONF_USERNAME
+from homeassistant.core import callback
+
+from .const import (  # pylint:disable=unused-import
+    CONF_CODE,
+    CONFIG_ENTRY_VERSION,
+    DOMAIN,
+)
+
+
+class HiveFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Hive config flow."""
+
+    VERSION = CONFIG_ENTRY_VERSION
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self.hive_auth = None
+        self.data = {}
+        self.tokens = {}
+        self.entry = None
+
+    async def async_step_user(self, user_input=None):
+        """Prompt user input. Create or edit entry."""
+        errors = {}
+        # Login to Hive with user data.
+        if user_input is not None:
+            self.data.update(user_input)
+            self.hive_auth = Auth(
+                username=self.data[CONF_USERNAME], password=self.data[CONF_PASSWORD]
+            )
+
+            # Get user from existing entry and abort if already setup
+            for entry in self._async_current_entries():
+                if (
+                    entry.data.get(CONF_USERNAME) == self.data[CONF_USERNAME]
+                    and not self.entry
```

Comment:
Before aborting for already configured unique id, check source. Ie if it's `SOURCE_REAUTH` we don't want to abort.